docs/products/opensearch/howto/get-started-with-nodejs.rst

```diff
@@ -0,0 +1,528 @@
+
+One of the examples for term-level query is searching for all entries with sodium value equal to 0.
+
+.. code-block:: javascript
+
+    /**
+     * Searching for exact matches of a value in a field.
+     */
+    module.exports.termSearch = (field, value) => {
+        console.log(`Searching for recipes with ${field} equal to ${value}`);
+        const body = {
+            'query': {
+                'term': {
+                    [field]: value
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body
+        }, logTitles)
+    };
+
+::
+
+    run-func index.js termSearch sodium 0
+
+
+Full-text queries
+^^^^^^^^^^^^^^^^^^
+
+Full-text queries returns results sorted by relevance. It allows higher flexibility to find most relevant results. For example, lets search for "Tomato garlic soup with dill" and look at the list of results.
+
+.. code-block:: javascript
+
+    /**
+     * Finding matches sorted by relevance.
+     */
+    module.exports.matchSearch = (field, query) => {
+        const body = {
+            'query': {
+                'match': {
+                    [field] : {
+                        query
+                    }
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body
+        }, logTitles)
+    };
+
+::
+
+    run-func index.js matchSearch title 'Tomato-garlic soup with dill'
+
+These are results which I got from the dataset:
+
+.. code-block:: text
+
+    [
+      'Mussel Soup with Avocado, Tomato, and Dill ',
+      'Garlic Soup ',
+      'Cucumber-Dill Soup with Scallions ',
+      'Cold Tomato-Thyme Soup with Grilled Garlic Croutons ',
+      'Chilled Tomato, Roasted Garlic, and Basil Soup ',
+      'Fast Favorite Garlic Dill Pickles ',
+      'Fast Favorite Garlic Dill Pickles ',
+      'Asparagus and Dill Avgolemono Soup ',
+      'Garlic Tomato Sauce ',
+      'Creamy Tomato Soup '
+    ]
+
+OpenSearch engine returned 10 most relevant results. Why 10? Because it is a default value. It can be increased by setting size property to a higher number.
+
+Next, lets use some special operators in a search query with *a query string*. Let's also increase the number of returned results to 100 to demonstrate how we can get more than 10 values.
+
+.. code-block:: javascript
+
+    /**
+     * Using special operators within a query string and a size parameter.
+     */
+    module.exports.querySearch = (query, size) => {
+        console.log(`Searching for ${query} and returning maximum ${size} results`);
+        const body = {
+            'query': {
+                'query_string': {
+                    "fields" : ["ingredients"],
+                    query
+                }
+            }
+        }
+        client.search({
+            index: indexName,
+            body,
+            size
+        }, logTitles)
+    };
+
```

Comment:
Again, let's add some words on which bits went in the body, and how we pass the size part separately so you could use that with any of the other query types too.